shared/actions/tracker.js

```diff
@@ -621,37 +531,118 @@ const _fillFolders = (username: string) => (dispatch: Dispatch, getState: () =>
     .filter(f => f.users.filter(u => u.username === username).length)
   dispatch(
     TrackerGen.createUpdateFolders({
-      username,
       tlfs,
+      username,
     })
   )
 }
 
-const updateTrackers = (username: string) => (dispatch: Dispatch, getState: () => TypedState) =>
-  Promise.all([listTrackers(username), listTracking(username)])
-    .then(([trackers, tracking]) => {
-      dispatch(TrackerGen.createSetUpdateTrackers({username, trackers, tracking}))
-    })
-    .catch(e => {
-      console.warn('Failed to get followers/followings', e)
-    })
+function* _updateTrackers(action: TrackerGen.UpdateTrackersPayload) {
+  const {username} = action.payload
+  try {
+    const [trackers, tracking] = yield Saga.all([
+      Saga.call(_listTrackersOrTracking, username, true),
+      Saga.call(_listTrackersOrTracking, username, false),
+    ])
+
+    yield Saga.put(TrackerGen.createSetUpdateTrackers({trackers, tracking, username}))
+  } catch (e) {
+    console.warn('Failed to get followers/followings', e)
+  }
+}
 
-const openProofUrl = (proof: Types.Proof) => (dispatch: Dispatch) => {
+function _openProofUrl(action: TrackerGen.OpenProofUrlPayload) {
+  const {proof} = action.payload
   openUrl(proof.humanUrl)
 }
 
-export {
-  getMyProfile,
-  getProfile,
-  onClose,
-  onFollow,
-  onIgnore,
-  onRefollow,
-  onUnfollow,
-  openProofUrl,
-  registerIdentifyUi,
-  setupUserChangedHandler,
-  startTimer,
-  triggerIdentify,
-  updateTrackers,
+function _userChanged(action: {payload: {uid: string}}, state: TypedState) {
+  const {uid} = action.payload
+  const actions = [TrackerGen.createCacheIdentify({goodTill: 0, uid})]
+  const username = _getUsername(uid, state)
+  if (username) {
+    actions.push(TrackerGen.createGetProfile({username}))
+  }
+  return actions
 }
+
+function _setupTrackerHandlers() {
+  engine().setIncomingActionCreators('keybase.1.NotifyUsers.userChanged', ({uid}) => {
+    return [{payload: {uid}, type: 'tracker:_userChanged'}]
+  })
+
+  engine().listenOnConnect('registerIdentifyUi', () => {
+    RPCTypes.delegateUiCtlRegisterIdentifyUIRpcPromise()
+      .then(response => {
+        console.log('Registered identify ui')
+      })
+      .catch(error => {
+        console.warn('error in registering identify ui: ', error)
+      })
+  })
+
+  // TODO get rid of getState here
+  engine().setIncomingActionCreators(
```

Comment:
i'm plumbing through dispatch and getState to this action creator temporarily. This is because this inner thing is using an incomingCallMap and not a nicer saga thing. this is the singular use of this